code/modules/media/broadcast/receivers/radio.dm

```diff
@@ -90,14 +90,83 @@
 	icon_state="wallradio"
 	anchored=1
 	volume=0.25 // 25% of user's set volume.
+	var/buildstage = 0
+
+/obj/machinery/media/receiver/boombox/wallmount/New(turf/loc,var/ndir=0,var/building=2)
+	..()
+	buildstage = building
+	if(!buildstage)
+		pixel_x = (ndir & 3)? 0 : (ndir == 4 ? 28 : -28)
+		pixel_y = (ndir & 3)? (ndir ==1 ? 28 : -28) : 0
+		dir=ndir
+		on = 0
+	update_icon()
 
 /obj/machinery/media/receiver/boombox/wallmount/update_on()
 	..()
-	if(on)
+	update_icon()
+
+/obj/machinery/media/receiver/boombox/wallmount/update_icon()
+	if(buildstage==2 && on)
 		icon_state="wallradio-p"
 	else
 		icon_state="wallradio"
 
+/obj/machinery/media/receiver/boombox/wallmount/attack_hand(var/mob/user)
+	if(buildstage<2)
+		return
+	else
+		return ..()
+
+/obj/machinery/media/receiver/boombox/wallmount/attackby(obj/item/weapon/W as obj, mob/user as mob)
+	switch(buildstage)
+		if(2)
+			if(iscrowbar(W))
+				user << "<span class='notice'>You pry the cover off [src].</span>"
+				playsound(get_turf(src), 'sound/items/Deconstruct.ogg', 50, 1)
+				if(do_after(user, src, 10))
```

Comment:
Putting do_after && buildstage = 2, would suffice as a busy check. Or you could be a real man and make a construction datum.